ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+    {
+        //if (camp->FieldDescription[0] + strlen(camp->FieldDescription[0]))
+        sprintf(camp->FieldDescription[0] + strlen(camp->FieldDescription[0]),
+                " (%u)", i);
+        for (i_camp = 0; i_camp < bd_xp->nFields; i_camp++)
+        {
+            if (bd_xp->pField + i_camp == camp)
+                continue;
+            if (!strcasecmp(bd_xp->pField[i_camp].FieldName, camp->FieldName))
+                break;
+        }
+        if (i_camp == bd_xp->nFields)
+            return retorn;
+    }
+    return 2;
+}  // End of MM_ModifyFieldNameAndDescriptorIfPresentBD_XP()
+
+static int MM_DuplicateMultilingualString(
+    char *(cadena_final[MM_NUM_IDIOMES_MD_MULTIDIOMA]),
+    const char *const(cadena_inicial[MM_NUM_IDIOMES_MD_MULTIDIOMA]))
+{
+    size_t i;
+
+    for (i = 0; i < MM_NUM_IDIOMES_MD_MULTIDIOMA; i++)
+    {
+        if (cadena_inicial[i])
+        {
+            if (nullptr == (cadena_final[i] = strdup(cadena_inicial[i])))
+                return 1;
+        }
+        else
+            cadena_final[i] = nullptr;
+    }
+    return 0;
+}
+
+int MM_DuplicateFieldDBXP(struct MM_FIELD *camp_final,
+                          const struct MM_FIELD *camp_inicial)
+{
+    *camp_final = *camp_inicial;
+
+    if (0 != MM_DuplicateMultilingualString(
+                 camp_final->Separator,
+                 (const char *const(*))camp_inicial->Separator))
+        return 1;
+
+    return 0;
+}
+
+char *MM_strnzcpy(char *dest, const char *src, size_t maxlen)
+{
+    size_t i = 0;
+    if (!src || maxlen == 0)
+    {
+        *dest = '\0';
+        return dest;
+    }
+
+    for (; i < maxlen - 1 && src[i] != '\0'; ++i)
+    {
+        dest[i] = src[i];
+    }
+    dest[i] = '\0';
+
+    return dest;
+}
+
+/*
+char *MM_strnzcpy(char *dest, const char *src, size_t maxlen)
+{
+    size_t i;
+    if (!src)
+    {
+        *dest = '\0';
+        return dest;
+    }
+
+    if (!maxlen)
+        i = 0;
+    else
+        strncpy(dest, src, i = maxlen - 1);
+
+    dest[i] = '\0';
+    return dest;
+}*/
+
+char *MM_oemansi_n(char *szcadena, size_t n_bytes)
+{
+    size_t u_i;
+    unsigned char *punter_bait;
+    unsigned char t_oemansi[128] = {
+        199, 252, 233, 226, 228, 224, 229, 231, 234, 235, 232, 239, 238,
+        236, 196, 197, 201, 230, 198, 244, 246, 242, 251, 249, 255, 214,
+        220, 248, 163, 216, 215, 131, 225, 237, 243, 250, 241, 209, 170,
+        186, 191, 174, 172, 189, 188, 161, 171, 187, 164, 164, 164, 166,
+        166, 193, 194, 192, 169, 166, 166, 164, 164, 162, 165, 164, 164,
+        164, 164, 164, 164, 164, 227, 195, 164, 164, 164, 164, 166, 164,
+        164, 164, 240, 208, 202, 203, 200, 180, 205, 206, 207, 164, 164,
+        164, 164, 166, 204, 164, 211, 223, 212, 210, 245, 213, 181, 254,
+        222, 218, 219, 217, 253, 221, 175, 180, 173, 177, 164, 190, 182,
+        167, 247, 184, 176, 168, 183, 185, 179, 178, 164, 183};
+    if (n_bytes == USHRT_MAX)
```

Comment:
It would probably be clearer to use SIZE_T_MAX here, and in the call to it from MM_oemansi(). And add a code comment to indicate it means that n_bytes is actually ignored then, and szcadena is assumed to be NUL terminated

Reply:
Perhaps you mean SIZE_MAX.
Done